offscreen/scripts/offscreen.ts

```diff
@@ -0,0 +1,15 @@
+import { MessageSender } from 'types/global';
+
+function responder(
+  message: any,
+  sender: MessageSender,
+  sendResponse: (response: any) => void,
+) {
+  console.log('message', message);
+  console.log('sender', sender);
+  sendResponse('acknowledged');
+}
+
+window.addEventListener('load', () => {
```

Comment:
Can we use post message streams?

Reply:
My thought was we just use the chrome.runtime.onMessage handler for the hardware wallets, but would that interfere with snaps setting up a PostMessageStream for the snaps communication? I am not very stream fluent, so if you could explain how it would work I'd be appreciative.

What I am hoping to do in layered prs is pull parts of @bergarces work in, here's an example:

https://github.com/MetaMask/metamask-extension/pull/17687/files#diff-5121fae02bbffe2a1d40923d2b6fa596ba8c4d41484a41027f1a1da95d472bbe

From: #17687